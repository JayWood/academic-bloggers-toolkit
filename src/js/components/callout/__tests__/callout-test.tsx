import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import * as React from 'react';

import Callout from '..';

describe('<Callout />', () => {
    it('should match snapshots', () => {
        let component = shallow(<Callout children="Hello world" />);
        expect(toJSON(component)).toMatchSnapshot();

        component = shallow(<Callout title="Hello!" children="Hello world" />);
        expect(toJSON(component)).toMatchSnapshot();

        component = shallow(<Callout children="" />);
        expect(toJSON(component)).toMatchSnapshot();

        component = shallow(
            <Callout intent="warning" children="Hello world" />,
        );
        expect(toJSON(component)).toMatchSnapshot();

        const dismiss = () => void 0;
        component = shallow(
            <Callout onDismiss={dismiss} children="Hello world" />,
        );
        expect(toJSON(component)).toMatchSnapshot();
    });
});
