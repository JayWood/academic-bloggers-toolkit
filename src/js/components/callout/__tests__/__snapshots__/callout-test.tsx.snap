// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Callout /> should match snapshots 1`] = `
<div
  aria-label="Error Hello world"
  className="callout danger"
  role="alert"
>
  <div
    className="heading"
  >
    <h5>
      Error
    </h5>
  </div>
  Hello world
</div>
`;

exports[`<Callout /> should match snapshots 2`] = `
<div
  aria-label="Hello! Hello world"
  className="callout danger"
  role="alert"
>
  <div
    className="heading"
  >
    <h5>
      Hello!
    </h5>
  </div>
  Hello world
</div>
`;

exports[`<Callout /> should match snapshots 3`] = `""`;

exports[`<Callout /> should match snapshots 4`] = `""`;

exports[`<Callout /> should match snapshots 5`] = `
<div
  aria-label="Warning Hello world"
  className="callout warning"
  role="alert"
>
  <div
    className="heading"
  >
    <h5>
      Warning
    </h5>
  </div>
  Hello world
</div>
`;

exports[`<Callout /> should match snapshots 6`] = `
<div
  aria-label="Error Hello world"
  className="callout danger"
  role="alert"
>
  <div
    className="heading"
  >
    <h5>
      Error
    </h5>
    <button
      aria-label="dismiss"
      className="dashicons dashicons-no-alt"
      onClick={[Function]}
    />
  </div>
  Hello world
</div>
`;
